app-android/src/main/java/com/ustadmobile/port/android/view/WorkspaceEnterLinkFragment.kt

```diff
@@ -0,0 +1,111 @@
+
+
+class WorkspaceEnterLinkFragment : UstadBaseFragment(), WorkspaceEnterLinkView{
+
+    private var mBinding: FragmentWorkSpaceEnterLinkBinding? = null
+
+    private var mPresenter: WorkspaceEnterLinkPresenter? = null
+
+    private val inputCheckDelay: Long = 250
+
+    private val inputCheckHandler: Handler = Handler()
+
+    private var lastInputTime: Long = 0
+
+    @VisibleForTesting
+    private var isIdle = false
+
+    private val inputCheckerCallback = Runnable {
+        if (System.currentTimeMillis() > lastInputTime + inputCheckDelay) {
+            mPresenter?.checkLinkValidity()
+        }
+    }
+
+    override var workspaceLink: String?
+        get() = mBinding?.workspaceLink
+        set(value) {}
+
+    override var validLink: Boolean = false
+        set(value) {
+            handleError(!value)
+            loading = false
+            mBinding?.showButton = value
+            field = value
+        }
+
+    override var progressVisible: Boolean = false
+        set(value) {
+            field = value
+            mBinding?.showProgress = value
+        }
+
+    private fun handleError(isError: Boolean){
+        mBinding?.workspaceLinkView?.isErrorEnabled = isError
+        mBinding?.workspaceLinkView?.error = if(isError) getString(R.string.invalid_url) else null
+    }
+
+    @TestOnly
+    private fun handleIdling(){
+        if(!isIdle){
+            isIdle = true
+            loading = isIdle
+        }
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
+                              savedInstanceState: Bundle?): View? {
+        val rootView: View
+        mBinding = FragmentWorkSpaceEnterLinkBinding.inflate(inflater, container, false).also {
+            rootView = it.root
+            it.showButton = false
+            it.showProgress = false
+        }
+        mPresenter = WorkspaceEnterLinkPresenter(requireContext(), UMAndroidUtil.bundleToMap(arguments),this)
+        mPresenter?.onCreate(savedInstanceState.toStringMap())
+        mBinding?.organisationLink?.addTextChangedListener(object: TextWatcher {
+            override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {}
+            override fun onTextChanged(url: CharSequence?, start: Int, before: Int, count: Int) {
+                inputCheckHandler.removeCallbacks(inputCheckerCallback)
+            }
+            override fun afterTextChanged(s: Editable?) {
+                val typedText = s.toString()
+                progressVisible = typedText.isNotEmpty()
+                val canValidate = PatternsCompat.WEB_URL.matcher(typedText).find()
+                        && typedText.contains(VALID_WORKSPACE_DOMAIN)
+                handleError(!canValidate)
+                if(!canValidate) return
+                handleIdling()
+                lastInputTime = System.currentTimeMillis()
+                inputCheckHandler.postDelayed(inputCheckerCallback, inputCheckDelay)
```

Comment:
How about we show the progress spinner as soon as we get afterTextChanged? Idling Resources could just watch for the progress bar.